backend/routes/seatbid.js

```diff
@@ -21,38 +19,60 @@ const router = express.Router();
 router.post('/start', authenticateUser, async (req, res) => {
     const { passengerId, seatNumber, flightId } = req.body;
     const biddingId = `${flightId}-${seatNumber}`;
-    const expirationTime = Date.now() + 30 * 60 * 1000; // 30 minutes
 
-    const existingBidding = Array.from(activeBiddings.values()).find(
-        bidding => bidding.passengerId === passengerId && bidding.status === 'ACTIVE'
-    );
+    try {
+        const flight = await prisma.flight.findUnique({ where: { flightId } });
+        if (!flight) {
+            return res.status(404).json({ error: 'Flight not found' });
+        }
 
-    if (existingBidding) {
-        return res.status(400).json({ error: 'You already have an active bidding' });
-    }
+        const departureDateTime = new Date(flight.scheduledDepartureTime);
 
-    const newBidding = {
-        passengerId,
-        seatNumber,
-        flightId,
-        startTime: Date.now(),
-        expirationTime,
-        status: 'ACTIVE'
-    };
+        if (isNaN(departureDateTime.getTime())) {
+            return res.status(400).json({ error: 'Invalid departure time' });
+        }
 
-    activeBiddings.set(biddingId, newBidding);
-    bids.set(biddingId, []);
+        const expirationTime = new Date(departureDateTime.getTime() - 60 * 60 * 1000); // 1 hour before departure
 
-    emitToFlight(req.app.locals.io, flightId, 'new-bid', { 
-        biddingId, 
-        seatNumber, 
-        passengerId,
-        bids: []
-    });
+        const existingBidding = await prisma.bidding.findFirst({
+            where: {
+                passengerId,
+                status: 'ACTIVE'
+            }
+        });
 
-    setTimeout(() => expireBidding(req.app.locals.io, biddingId), expirationTime - Date.now());
+        if (existingBidding) {
+            return res.status(400).json({ error: 'You already have an active bidding' });
+        }
 
-    res.json({ success: true, biddingId, ...newBidding, bids: [] });
+        const newBidding = await prisma.bidding.create({
+            data: {
+                biddingId,
+                flightId,
+                passengerId,
+                seatNumber,
+                startTime: new Date(),
+                expirationTime,
+                status: 'ACTIVE'
+            }
+        });
+
+        emitToFlight(io, flightId, 'new-bid', { 
+            biddingId, 
+            seatNumber, 
+            passengerId,
+            startTime: new Date(),
+            expirationTime,
+            bids: []
+        });
+
+        setTimeout(() => expireBidding(io, biddingId), expirationTime.getTime() - Date.now());
```

Comment:
What would happen in server is killed and this code cannot be executed? Can we evaluate whether bidding has expired every time we return active biddings from database?

Reply:
You are right. That case isn't handled. Let's chat about it more later today!